src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISExpressDeployer.cs

```diff
@@ -295,6 +295,12 @@ private void PrepareConfig(string contentRoot, int port)
             }
             else
             {
+                if (IISDeploymentParameters.WebConfigBasedEnvironmentVariables.Count > 0)
```

Comment:
I would prefer fixing logic inside RunWebConfigActions and calling it outside the if.